### PR TITLE
Add more subtests for autotest.pm

### PR DESCRIPTION
**Pull request by gauravpathak**
add unit tests to have more code coverage

### CODE REVIEW

File: t/08-autotest.t [hunk 1]
```diff
@@ -172,6 +172,7 @@ subtest 'test always_rollback flag' => sub {
     snapshot_subtest 'stopping overall test execution early due to fatal test failure' => sub {
         $mock_basetest->redefine(runtest => sub { die "test died\n" });
         $vm_stopped = 0;
+        $bmwqemu::vars{DUMP_MEMORY_ON_FAIL} = 'true';
```
Comment (Martchus) [inline t/08-autotest.t:175]: We usually just use 0 or 1 for boolean variables in Perl:
```suggestion
        $bmwqemu::vars{DUMP_MEMORY_ON_FAIL} = 1;
```

**Review by Martchus — COMMENTED** [review]
Looks generally good.

File: t/08-autotest.t [hunk 2]
```diff
@@ -422,6 +424,30 @@ subtest rollback_activated_consoles => sub {
     is_deeply(\@selected_consoles, [{cmd => 'backend_select_console', testapi_console => 'last_milestone_console'}], 'last milestone console selected');
 };
 
+subtest find_script => sub {
+    my $override_path = 0;
+    $bmwqemu::vars{WHEELS_DIR} = '';    # overwrite WHEELS_DIR and set it empty
+    $bmwqemu::vars{ASSETDIR} = Cwd::getcwd . '/t/data/assets';
```
Comment (Wabri) [inline t/08-autotest.t:430]: Why not use the start.pm fake data? `t/fake/tests/start.pm`

Comment (gauravpathak) [reply t/08-autotest.t:430]: This condition is actually covering these lines: https://github.com/os-autoinst/os-autoinst/blob/master/autotest.pm#L56-L58
To cover `$script_override_path‎`

**Review by okurz — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Pull request merged**